lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1994,13 +2027,14 @@ static void checkAccessControl(TypeChecker &TC, const Decl *D) {
           return;
 
         checkTypeAccess(TC, TypeLoc::withoutLoc(theVar->getType()),
-                        theVar,
+                        theVar, /*checkUsableFromInline=*/false,
                         [&](AccessScope typeAccessScope,
                             const TypeRepr *complainRepr,
                             DowngradeToWarning downgradeToWarning) {
           auto typeAccess = typeAccessScope.accessLevelForDiagnostics();
           bool isExplicit =
-            theVar->getAttrs().hasAttribute<AccessControlAttr>();
+            theVar->getAttrs().hasAttribute<AccessControlAttr>() ||
+            isa<ProtocolDecl>(theVar);
```

Comment:
Definitely needs a new test if existing ones don't cover it.

Reply:
See https://github.com/apple/swift/pull/16586/files#diff-8ace5fe1c031aa385557a1c025b63437, I think I covered it